app/sentinel/background/initialization/validation.go

```diff
@@ -11,46 +11,36 @@
 package initialization
 
 import (
-	"fmt"
-	"os"
-	"strings"
-
-	"github.com/vmware-tanzu/secrets-manager/core/env"
+	"context"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"github.com/vmware-tanzu/secrets-manager/app/sentinel/internal/safe"
+	"github.com/vmware-tanzu/secrets-manager/core/backoff"
 	log "github.com/vmware-tanzu/secrets-manager/core/log/std"
 )
 
-func initCommandsExecutedAlready(cid *string) bool {
-	log.TraceLn(cid, "checking tombstone file")
+func initCommandsExecutedAlready(ctx context.Context, src *workloadapi.X509Source) bool {
+	cid := ctx.Value("correlationId").(*string)
 
-	// Parse tombstone file first:
-	tombstonePath := env.InitCommandTombstonePathForSentinel()
-	file, err := os.Open(tombstonePath)
-	if err != nil {
-		log.InfoLn(
-			cid,
-			"RunInitCommands: no tombstone file found... skipping custom initialization.",
-		)
-		return false
-	}
+	log.TraceLn(cid, "check:initCommandsExecutedAlready")
+
+	s := backoffStrategy()
+	initialized := false
+	for {
```

Comment:
The reason we use infinite loops is because you will never know how long it will take a system to reconcile.

VSecM deploys very fast, but if there is resource contention due to cluster initialization, it might need to for a while for the workload api to be available.

Since this “a while” is rather vague, and since things will reconcile eventually, we use infinite loops with exponential backoffs.

Another alternative would have been to use a message queue. but then
1. we’d need to guarantee that the queue has reconciled.
2. it would make the code more complicated.
3. we’d depend on yet another thing.
4. adding a message queue would mean a larger footprint for the deployment.